core/src/main/java/org/apache/james/core/Domain.java

```diff
@@ -27,6 +27,14 @@
 public class Domain {
 
     public static final Domain LOCALHOST = Domain.of("localhost");
+    public static final Domain NULL = new Domain("");
```

Comment:
what is it for?

Reply:
In SMTP envelope, you can get a null sender. That, for instance is a valid SMTP command:

```
MAIL FOM: <>
```

James hooks on MailAddress parsing to handle that case at the SMTP level.

I guess the right representation of that is to have the sender part of the envelope being optional...

Reply:
I added tests for `<>` sender support
 - Was broken on MailAddress move to `core`
 - I spotted uneeded MailAddress to MailAddress conversion (relique of the past when SMTP and processing used to have two different objects linked by an adapter)